NewForce-Cohort-5/tabloidmvc-the-spaghetti-squad
# AE Ticket #3 - View All Posts

# Description

This is a pull request for the [ticket #3](https://trello.com/c/66nmZoxV/3-3-view-all-posts) to view all posts at the Posts list page after a user logs in and selects the `Posts` from the menu option.   Each post in the list should display the title, author and category.  Each post should display PublishDateTime.  All post should have the PublishDateTime of the past and should be displayed in the descending order by PublishDateTime (i.e. the most recent one on the top).  Lastly only approved posts should be displayed.  

Related Ticket(s):  
[https://trello.com/c/66nmZoxV/3-3-view-all-posts](https://trello.com/c/66nmZoxV/3-3-view-all-posts)

Fixes # (issue): The list was not displayed in the descending order by PublishDateTime.  So I updated the already exiting variable called post inside the index method in the PostController.cs file to display the post by the descending order by PublishDateTime.

## Type of change

Please delete options that are not relevant.





# Testing Instructions
**testing objectives**
1. to see if the post in the list should display the title, author and category.  Each post should display PublishDateTime
2. to see if the past post will be display in the descending order by the publish date time 
3. to see if the future post will not displayed
4. to see if the non approved post will not be displayed 

**testing instructions**
1. In Terminal, cloning down this repository to your local repository.  
2. If you currently do not have access to this branch, run a <git fetch --all>, then to navigate to this branch `ae-post-view` by typing `git checkout ae-post-view` in Terminal.  And type `Start TabloidMVC.sln` to run this program through Visual Studio 2022.  
3. If you have not created or seeded the basic database, you will need to do so.  First, open a new query by right click on TabloidMVC under the database file in the SQL Sever Object Explorer in the left.  
4. You will need to create the table first and then add the seed data. Choose files in the following order:
5. [01_TabloidMVC_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-spaghetti-squad/blob/main/SQL/01_TabloidMVC_Create_DB.sql)
6. [02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-spaghetti-squad/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)
7. To run the query, please click the arrow under the query tab.
8. Next, we need to add more data to the database to test the aforementioned testing objectives. 
9. Please past the following sql commands at the query for the TabloidMVC database for the testing objective 2 (to see if the past posts will be display in the descending order by the publish date time) and click the arrow under the query tab.  This step is to add the data at this point. 
```
Insert into Post 
(Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId )
VALUES ('SQl is the second best', 'How do you feel about SQL', 'https://ih1.redbubble.net/image.471342622.5626/raf,750x1000,075,t,101010:01c5ca27c6.u4.jpg', '2021-01-02 10:35:00.830','2021-03-02 10:35:00.830', 1, 1, 1), ( 'JavaScript', 'Do you remember JavaScript?', 'https://miro.medium.com/max/2560/0*ljnWAuUhEYej-FQ2.jpeg', '2021-12-02 08:06:21.354', '2021-12-02 08:06:21.354', 1, 1, 1 ),  ('CSS', 'Graphic is important', 'https://www.logolynx.com/images/logolynx/8c/8cdf4c047e99f7389e76aa4e2e7e2803.png', '2021-10-28 08:06:21.354', '2021-10-28 08:06:21.354', 1, 1, 1 );

select * from Post;
```
10. Please past the following sql commands at the query for the TabloidMVC database for the testing objective 3 (to see if the future post will not displayed) and click the arrow under the query tab.  This is to add the data at this point. 
```
Insert into Post 
(Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId )
VALUES ('Web API is the future', 'Are you ready for Web API and ASP.NET Core', 'https://i.morioh.com/2019/11/21/c33e0110f8bf.jpg', '2023-01-02 10:35:00.830','2023-03-02 10:35:00.830', 1, 1, 1);

select * from Post;
```
11. Please past the following sql commands at the query for the TabloidMVC database for checking the testing objective 4 ( to see if the non approved post will not be displayed) and click the arrow under the query tab.  This is to add the data at this point.  
```
 Insert into Post 
(Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId )
 VALUES ('Not approved', 'This is a test post for not approved', 'https://st2.depositphotos.com/1031343/7519/v/950/depositphotos_75193785-stock-illustration-not-approved-stamp.jpg', '2022-03-02 13:15:45.730','2022-03-02 13:15:45.730', 0, 1, 1);

select * from Post;
```

12. Thank you for adding the data.  To check the aforementioned testing objectives, please click the solid green debugger play button next to IISExpress.
13. A browser should start to run the program with the localhost as a part of the URL.  The opening page with Tabloid and Menu in the heading should be displayed.  Please scroll down to see the Log In button.  Please click the button.
11. The log-in page should be displayed.  Please type the following user email of "[admin@example.com](mailto:admin@example.com)" in the input field and click the submit button. 
12. The home page, which is similar to the log-in page, should be displayed.  At this page, you should not see the log-in button.  
13.  Please select the `Posts` from the menu option at top right.
14.  The page displays a list of posts.  Please check the followings:
A. testing objective 1:  to see if the posts in the list should display the title, author and category.  Each post should display PublishDateTime under the published column;
B. testing objective 2: to see if the past post will be display in the descending order by the publish date time 
C. testing objective3:  to see if the future post will not displayed.  Make sure that a post title "Web API is the future" should not displayed. 
D. testing objective 4: to see if the non approved post will not be displayed.  Make sure that a post title "Not Approved" should not displayed.  




# Checklist:
